### ISSUE TITLE
Mac OSX binaries for v1.1.0-beta.1 require OSX 10.12

### ISSUE THREAD

**Issue by psibre** [MontrealCorpusTools/Montreal-Forced-Aligner#95]
Unlike MFA v1.0.0, where the Mac OSX binaries worked on OSX 10.10 and above, the binaries for v1.1.0-beta.1 do not work with OSX 10.11 and below.
Specifically, with v1.1.0-beta.1, this works on OSX 10.12 and 10.13:
```
$ lib/align -h
usage: align [-h] [-s SPEAKER_CHARACTERS] [-t TEMP_DIRECTORY] [-j NUM_JOBS]
             [-v] [-c] [-d] [--config_path CONFIG_PATH]
             corpus_directory dictionary_path acoustic_model_path
             output_directory

positional arguments:
  corpus_directory      Full path to the directory to align
  dictionary_path       Full path to the pronunciation dictionary to use
  acoustic_model_path   Full path to the archive containing pre-trained model
                        or language (english)
  output_directory      Full path to output directory, will be created if it
                        doesn't exist

optional arguments:
  -h, --help            show this help message and exit
  -s SPEAKER_CHARACTERS, --speaker_characters SPEAKER_CHARACTERS
                        Number of characters of file names to use for
                        determining speaker, default is to use directory names
  -t TEMP_DIRECTORY, --temp_directory TEMP_DIRECTORY
                        Temporary directory root to use for aligning, default
                        is ~/Documents/MFA
  -j NUM_JOBS, --num_jobs NUM_JOBS
                        Number of cores to use while aligning
  -v, --verbose         Print more information during alignment
  -c, --clean           Remove files from previous runs
  -d, --debug           Output debug messages about alignment
  --config_path CONFIG_PATH
                        Path to config file to use for alignment
```
But on OSX 10.11, this happens:
```
$ lib/align -h
[65846] Error loading Python lib '/Users/ingmar/Downloads/montreal-forced-aligner_macosx/lib/Python': dlopen: dlopen(/Users/ingmar/Downloads/montreal-forced-aligner_macosx/lib/Python, 10): Symbol not found: _clock_getres
  Referenced from: /Users/ingmar/Downloads/montreal-forced-aligner_macosx/lib/Python (which was built for Mac OS X 10.12)
  Expected in: /usr/lib/libSystem.B.dylib
 in /Users/ingmar/Downloads/montreal-forced-aligner_macosx/lib/Python
```
This seems to be unintended, and the docs currently state that [it should work on OSX 10.11](https://montreal-forced-aligner.readthedocs.io/en/latest/installation.html#mac).

Unless there is a compelling reason why MFA v1.1.0 cannot support OSX 10.11, it would be great if the Mac OSX binaries could be built to be compatible with OSX 10.11 (and maybe even 10.10?).

**Comment by psibre** [comment]
Incidentally, we integrate MFA in a Gradle plugin, which is currently tested on Travis CI using this build matrix:
```yaml
matrix:
  include:
  - os: linux
    jdk: oraclejdk8
  - os: osx # OSX 10.10
    osx_image: xcode6.4
  - os: osx # OSX 10.11
    osx_image: xcode8
  - os: osx # OSX 10.12
    osx_image: xcode9
  - os: osx # OSX 10.13
```
With MFA v1.0.0, [everything works very nicely](https://travis-ci.org/marytts/gradle-marytts-kaldi-mfa-plugin/builds/430003897). But with MFA v1.1.0-beta.1, [the OSX 10.10 and 10.11 builds fail](https://travis-ci.org/marytts/gradle-marytts-kaldi-mfa-plugin/builds/430019621).

**Comment by mmcauliffe** [comment]
Yeah, the build machine I use for Mac got rebuilt with a newer OS earlier this year.  I don't think I have access to a build machine for 10.10 anymore, so I'll have to update the docs.  There are some lab mac laptops that might have older versions that I could use, so I'll double check in a week or so before committing to updating the Mac OS version requirments.

**Comment by psibre** [comment]
Thanks for the quick response and insight! I don't know exactly how the binaries are compiled -- maybe there are flags to make it backward-compatible?
Depending on how complicated the build is, you might also [use Travis to build and deploy](https://docs.travis-ci.com/user/deployment/releases/). But last time I build MFA from source (early 2016, iirc), it was a significant pain to get Kaldi built correctly. There might be ways to work around that, but that's another discussion.

On that note, even if you prefer to build the binaries locally, let me know if there's anything I can do to help. I have OSX machines going back to 10.7, as well as VMware Fusion for vanilla sandboxes.

**Comment by mmcauliffe** [comment]
Right, so I moved the building of kaldi/opengrm-ngram/phonetisaurus offline and have it download prebuilt versions of those for travis, simply because the time it takes to compile kaldi means that testing times out travis.  I could do some independent compiling in different repos and then use those builds for distribution, at least for Mac and Ubuntu.  I'll try to get this better solved for 1.1.
